dialogs/measurement/depth_profile_numeric_limits.py

```diff
@@ -0,0 +1,73 @@
+# coding=utf-8
+"""
+Created on 21.3.2013
+Updated on 30.5.2018
+
+Potku is a graphical user interface for analyzation and
+visualization of measurement data collected from a ToF-ERD
+telescope. For physics calculations Potku uses external
+analyzation components.
+Copyright (C) 2013-2018 Jarkko Aalto, Severi Jääskeläinen, Samuel Kaiponen,
+Timo Konu, Samuli Kärkkäinen, Samuli Rahkonen, Miika Raunio, Heta Rekilä and
+Sinikka Siironen, 2020 Juhani Sundell, Tuomas Pitkänen
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program (file named 'LICENCE').
+"""
+__author__ = "Jarkko Aalto \n Timo Konu \n Samuli Kärkkäinen" \
+             " \n Samuli Rahkonen \n Miika Raunio \n Severi Jääskelänen \n " \
+             "Samuel Kaiponen \n Heta Rekilä \n Sinikka Siironen" \
+             "Juhani Sundell \n Tuomas Pitkänen"
+__version__ = "2.0"
+
+import widgets.binding as bnd
+import widgets.gui_utils as gutils
+
+from PyQt5 import QtCore
+from PyQt5 import uic
+from PyQt5 import QtWidgets
+
+
+class NumericLimitsDialog(QtWidgets.QDialog):
+    """Numeric limits dialog for the depth profile graph.
+    """
+    color_scheme = bnd.bind("colorbox")
+
+    def __init__(self, parent):
+        """Inits Depth profile numeric limits dialog.
+        
+        Args:
+            parent: MatplotlibDepthProfileWidget which limits are being changed.
+        """
+        super().__init__()
+        uic.loadUi(gutils.get_ui_dir() / "ui_depth_profile_limits.ui", self)
+        self.setAttribute(QtCore.Qt.WA_DeleteOnClose)
+
+        self.parent = parent
```

Comment:
I am not a fan of child objects having references to their parent objects and I especially do not like child objects modifying their parents like on lines 70-71. I know this is common in Potku's source code, but we should aim to get rid of code like this.

For a better way to implement this type of dialog, check out [DepthProfileIgnoreElements](https://github.com/JYU-IBA/potku/blob/master/dialogs/measurement/depth_profile_ignore_elements.py) and how it is used in [DepthProfileWidget](https://github.com/JYU-IBA/potku/blob/master/widgets/matplotlib/measurement/depth_profile.py#L388-L400).